Adds section on bundle definition file and removes Docker refs in mixer (#1050)
clearlinux/clear-linux-documentation: source/guides/clear/mixer.rst

used on a bare metal system instead of a VM. So we will modify the default
bundle set to get a smaller kernel image, which will also be faster to load.

.. note:: 
   
   The only bundles available to :command:`swupd` for a given release are 
   those that were added to the mix during build time. A mix doesn’t 

>    automatically inherit upstream bundles.

Reviewer: ```suggestion
   automatically inherit upstream all bundles.
```

Author: Done.

used on a bare metal system instead of a VM. So we will modify the default
bundle set to get a smaller kernel image, which will also be faster to load.

In this example, we introduce two type of bundles: ``package bundles`` that
contain the list of RPMs; and ``content bundles`` that contain the path to 
the local content. We recommend that the local content resides outside the 

> mixer work space.   

Author: Add a line: 

Mixer now supports creation of third-party bundles out of any content you want without the need to package it into an RPM format first.  Refer to `swupd`_ [add link]

      mixer bundle add editors
      mixer bundle edit editors

#. We can build a mix from RPM packages (a.k.a ``package bundles``) 
   as well as from ``content bundles``. For building a mix from RPM
   packages, specify the package name in the bundle definition file, as
   shown below. 

   .. code-block:: bash

      TBD

#. For building a mix from ``content bundles``, specify the file 
   path to the content in the bundle definition file, using the keyword 

>    :gui:`content()`. See the example below. 

Reviewer 2: I think you mean :guilabel: not :gui: (line 301).

Author: Great catch. Thank you.

   #builder.conf

   #VERSION 1.2

   [Builder]
      CERT = "/home/clr/mix/Swupd_Root.pem"
      SERVER_STATE_DIR = "/home/clr/mix/update"
      VERSIONS_PATH = "/home/clr/mix"
      YUM_CONF = "/home/clr/mix/.yum-mix.conf"

   [Swupd]

>       BUNDLE = "os-core-update"

Author: In my `builder.conf`, Ln 740 shows:
`BUNDLE = "os-core" [**not** "os-core-update" Please advise which is correct. ]

Reviewer: it should be "os-core-update".  You're probably seen "os-core" because that is the configuration for build 3rd-party bundle.  If you just start another mix from scratch, you should see "os-core-update".

Author: Thanks for the clarification.

      CONTENTURL = "<URL where the content will be hosted>"
      VERSIONURL = "<URL where the version of the mix will be hosted>"
      COMPRESSION = ["external-xz"]
      UPSTREAM_BUNDLES_URL = "https://github.com/clearlinux/clr-bundles/archive/"

   [Server]
      DEBUG_INFO_BANNED = "true"
      DEBUG_INFO_LIB = "/usr/lib/debug"
      DEBUG_INFO_SRC = "/usr/src/debug"

   [Mixer]
      LOCAL_BUNDLE_DIR = "/home/clr/mix/local-bundles"
      LOCAL_REPO_DIR = "/home/clr/mix/local-yum"

>       LOCAL_RPM_DIR = "/home/clr/mixx/local-rpms"

Author: Is **mixx** a typo? On Ln 754, why does the value
`LOCAL_RPM_DIR =  "/home/clr/mixx/local-rpms"` ?

Reviewer: you're right, it should be "mix", not "mixx".

Author: Cool. See latest commit.

>    #VERSION 1.0

Reviewer: ```suggestion
   #VERSION 1.2
```

     LOCAL_BUNDLE_DIR = "/home/clr/mix/local-bundles"
     LOCAL_REPO_DIR = ""
     LOCAL_RPM_DIR = ""

>      DOCKER_IMAGE_PATH = "clearlinux/mixer" [REPLACEMENT?]

Author: @bktan8 Will you please confirm that I can completely remove line 752, above? 
My only concern was whether this removal required a REPLACEMENT.

Reviewer: @mvincerx - you can replace it with this:

```
#VERSION 1.2

[Builder]
  CERT = "/home/clr/mix/Swupd_Root.pem"
  SERVER_STATE_DIR = "/home/clr/mix/update"
  VERSIONS_PATH = "/home/clr/mix"
  YUM_CONF = "/home/clr/mix/.yum-mix.conf"

[Swupd]
  BUNDLE = "os-core-update"
  CONTENTURL = "<URL where the content will be hosted>"
  VERSIONURL = "<URL where the version of the mix will be hosted>"
  COMPRESSION = ["external-xz"]
  UPSTREAM_BUNDLES_URL = "https://github.com/clearlinux/clr-bundles/archive/"

[Server]
  DEBUG_INFO_BANNED = "true"
  DEBUG_INFO_LIB = "/usr/lib/debug"
  DEBUG_INFO_SRC = "/usr/src/debug"

[Mixer]
  LOCAL_BUNDLE_DIR = "/home/clr/mix/local-bundles"
  LOCAL_REPO_DIR = "/home/clr/mix/local-yum"
  LOCAL_RPM_DIR = "/home/clr/mixx/local-rpms"
  OS_RELEASE_PATH = ""
```

Author: @bktan8 See my latest commmit, below.  Please note: 
- Not every `Variable` is explained in the table (This was true prior to this PR)
- In case you want each Variable to be explained, please LMK.

Reviewer: @mvincerx - I noticed the `#VERSION` in the new doc is still 1.0, but it should be 1.2.

Author: I corrected `#VERSION` to 1.2.

precedence enables you to copy upstream bundles locally, and edit into a
local variation.

Bundle definition files
-----------------------

A ``bundle definition`` file consists of a header, followed by a list 
of packages and directives. The header holds important meta-data, like 
the TITLE, DESCRIPTION, and STATUS. Other meta-data include TAGS, which 
define a bundle's function in the ecosystem, and MAINTAINER, which gives 
contact information. 

Following the header are the directives, shown in Table 2. 

.. list-table:: **Table 2**: Bundle directives
   :widths: 50,50
   :header-rows: 1 

   * - **Directive**

>      - **Explanation**

Reviewer: ```suggestion
     - **Description**
```

Author: I'm OK with changing this Column header to "Description". However, if we change this, then the Column header in Table 1, "Explanation", should also be changed.  I changed the latter.

   * - ``include(<required-bundle-name>)``

>      - Add separate bundle with current bundle

Reviewer: ```suggestion
     - Add <required-bundle-name> bundle with this bundle
```

>    * - ``also-add(<required-bundle-name>)``

Reviewer: ```suggestion
   * - ``also-add(<optional-bundle-name>)``
```

>      - Add with this bundle unless the ``swupd bundle-add`` option   ``--skip-optional`` is used.

Reviewer: ```suggestion
     - Add <optional-bundle-name> with this bundle unless the ``swupd bundle-add`` option   ``--skip-optional`` is used.
```

Author: See commit. Sentence needs simplification.

   * - ``content(<full/path/to/non-packaged/content>)``

>      - Add the non-packaged content to the bundle. This directive is explained in :ref:`swupd-3rd-party`

Reviewer: ```suggestion
     - Add the non-packaged content to the bundle. The usage of this directive is demonstrated in :ref:`swupd-3rd-party`
```

Author: See commit. This sentence requires simplification.

Following is `cluster-tools`, an upstream bundle definition file, where

> directives are highlighted.

Reviewer: ```suggestion
directives are highlighted and the rest are packages.
```

Author: See commit.
